pkg/operator/targetconfigcontroller/targetconfigcontroller.go

```diff
@@ -246,3 +264,38 @@ func (c *TargetConfigController) namespaceEventHandler() cache.ResourceEventHand
 		},
 	}
 }
+
+// ensureCSRControllerCA validates that the openshift-config-managed configmap csr-controller-ca contains a valid
+// CA signed by kubelet-signer. This is required to ensure rollout happens in a time that minimizes possible
+// cluster disruption in non HA deployments.
+func ensureCSRControllerCAConfigMap(csrControllerCAConfigMap *corev1.ConfigMap) error {
+	caBundle := csrControllerCAConfigMap.Data["ca-bundle.crt"]
+	if len(caBundle) == 0 {
+		return fmt.Errorf("configmap openshift-config-managed/csr-controller-ca is not valid, waiting on kube-controller-manager-operator to create")
+	}
+	block, _ := pem.Decode([]byte(caBundle))
+	leafCert, err := x509.ParseCertificate(block.Bytes)
+	if err != nil {
+		return err
+	}
+	if len(leafCert.Subject.CommonName) == 0 {
+		return fmt.Errorf("configmap openshift-config-managed/csr-controller-ca is not valid, waiting on kube-controller-manager-operator to generate")
+	}
+
+	isCASubjectExpected := strings.HasPrefix(leafCert.Subject.CommonName, "kube-csr-signer")
```

Comment:
this might be too brittle but I added it as another validation signal as manager might not be enough? we could consider additional annotations or labels on the configmap as well.

Reply:
Hmm... I wonder if we couldn't just check that CM's `.metadata.managedFields.manager` instead, if last to touch it is `cluster-kube-controller-manager-operator` it means we got the reaction we cared about. Before the manager will be `cluster-kube-apiserver-operator` iirc, right?